src/sidebar/components/menu-keyboard-navigation.js

```diff
@@ -8,27 +8,45 @@ function isElementVisible(element) {
   return element.offsetParent !== null;
 }
 
+/**
+ * @template T
+ * @typedef {import("preact/hooks").Ref<T>} Ref
+ */
+
+/**
+ * @typedef MenuKeyboardNavigationProps
+ * @prop {string} [className]
+ * @prop {(e: KeyboardEvent) => any} [closeMenu] - Callback when the menu is closed via keyboard input
+ * @prop {boolean} [visible] - When  true`, sets focus on the first item in the list
+ * @prop {Object} children - Array of nodes which may contain <MenuItems> or any nodes
+ */
+
 /**
  * Helper component used by Menu and MenuItem to facilitate keyboard navigation of a
  * list of <MenuItem> components. This component should not be used directly.
  *
  * Note that `ArrowRight` shall be handled by the parent <MenuItem> directly and
  * all other focus() related navigation is handled here.
+ *
+ * @param {MenuKeyboardNavigationProps} props
  */
 export default function MenuKeyboardNavigation({
   className,
   closeMenu,
   children,
   visible,
 }) {
-  const menuRef = useRef(null);
+  const menuRef = /** @type {Ref<HTMLDivElement|null>} */ (useRef(null));
 
   useEffect(() => {
     let focusTimer = null;
     if (visible) {
       focusTimer = setTimeout(() => {
         // The focus won't work without delaying rendering.
-        const firstItem = menuRef.current.querySelector('[role^="menuitem"]');
+        /** @type {HTMLFormElement|null} */
+        const firstItem = /** @type {Ref<HTMLDivElement>} */ (menuRef).current.querySelector(
```

Comment:
Still needing to cast everywhere since it does not know this is not null :\ maybe there is a better way to do this?

Reply:
See my note above about specifying the type by casting the `useRef` argument.